### DISCUSSION TITLE
add support for multi model combobox?

### DISCUSSION THREAD

**Discussion by buncis** [josefarias/hotwire_combobox#48] [Ideas]
kinda like grouped collection select, but not it 

in my current use case, I have a model with polymorphic `belongs_to`

so on my current typeahead combobox implementation

I have one field to show the selections of models, and one fields to select certain record of that models

**Comment by josefarias** [comment]
Hey @buncis! Thanks for the suggestion.

My current vision for something like this is that you should architect your application with either single-table inheritance or delegated types, such that you can pass all options in a single activerecord relation to the combobox, across different models.

Is there any reason this wouldn't work for you? If so, happy to consider other approaches. An API proposal would be good, though, as I'm not able to picture a multi-model one, currently.

Once we support grouped collections, which I do want to do eventually, we can match on the type column of the relation to group the options.

**Reply by buncis** [reply]
at my use case sti would give more problems than using polymorphic,

eg: the table would be really bloated with many of them have an empty columns, "
and the record number would be exploded 
(idk what correct word to choose, pardon)

because its really a different model/domain 
and have separated concern instead of just a variant 
thus why I go with polymorphic


this `delegated types` are new to me, I'm looking at it currently

**Reply by josefarias** [reply]
Ah yeah that seems like a very good use case for delegated types. Check it out and let me know if it doesn't work for you. I hope it does work! 👍

**Reply by buncis** [reply]
apparently its been there since rails 6.1, and it written by dhh himself https://github.com/rails/rails/pull/39341, wish I know about it earlier 

I think it will work with delegate types, I just need do to some migration and refactor at my side XD
